src/cluster/src/helm.rs

```diff
@@ -0,0 +1,141 @@
+use std::io::Error as IoError;
+use std::io::ErrorKind;
+use std::process::{Command, Stdio};
+
+use serde::Deserialize;
+use flv_util::cmd::CommandExt;
+
+#[derive(Debug)]
+#[non_exhaustive]
+pub struct HelmClient {}
+
+impl HelmClient {
+    /// Creates a Rust client to manage our helm needs.
+    ///
+    /// This only succeeds if the helm command can be found.
+    pub fn new() -> Result<Self, IoError> {
+        let output = Command::new("helm").arg("version").print().output()?;
+
+        // Convert command output into a string
+        let out_str = String::from_utf8(output.stdout).map_err(|e| {
+            IoError::new(
+                ErrorKind::InvalidData,
+                format!("failed to parse helm output as string: {}", e),
+            )
+        })?;
+
+        // Check that the version command gives a version.
+        // In the future, we can parse the version string and check
+        // for compatible CLI client version.
+        if !out_str.contains("version") {
+            return Err(IoError::new(
+                ErrorKind::InvalidData,
+                "failed to get helm version",
+            ));
+        }
+
+        // If checks succeed, create Helm client
+        Ok(Self {})
+    }
+
+    /// Installs the given chart under the given name.
+    ///
+    /// The `opts` are passed to helm as `--set` arguments.
+    pub fn install(
+        &self,
+        namespace: &str,
+        name: &str,
+        chart: &str,
+        opts: &[(&str, &str)],
+    ) -> Result<(), IoError> {
+        let mut sets = vec![];
+        for (key, val) in opts {
+            sets.push("--set".to_string());
+            sets.push(format!("{}={}", key, val));
+        }
+
+        let status = Command::new("helm")
+            .args(&["install", name, chart])
+            .args(&["-n", namespace])
+            .args(sets)
+            .stdout(Stdio::inherit())
+            .stderr(Stdio::inherit())
+            .print()
+            .status()?;
```

Comment:
The problem I have with the `inherit()` extension is that it will panic if the executable fails. As a library component, this should not panic, but return an error type describing the problem. It seems to me like `inherit()` does too many things all at once, it would be nice if it just set the pipes as inherited and then gave back the `&mut Command` handle like the rest of the Command functions do.

Reply:
whatever is ok, as long as following conditions are met:
- if error occurs, error message should be returned not just status
- use common extension so no copy and paste

Reply:
So recently, we added `CommandExt` to the `flv_util` crate in the `cmd` module. That is brand new in terms of usage, and I believe the goal was to eventually migrate other usages of `CommandUtil` to use that. What if I changed the `inherit()` method on the new `CommandExt` to just do the forwarding and pass-through that I described? Since nothing is ported over to that yet it wouldn't really be a breaking change, then we can slowly adjust the other usages to adopt this new pattern. How does that sound?

Reply:
sounds good.  just make sure we got error back which was problem trying to debug problem

Reply:
Proposed changes still doesn't keep old behavior of `inherit` which prints out error only if error occurs.  For this PR, we should keep existing behavior as is.  Changes to CommandExt and related changes should be made as separate PR